src/classes/ADDR_Account_TDTM.cls

```diff
@@ -48,123 +51,122 @@ public class ADDR_Account_TDTM extends TDTM_Runnable {
             
         DmlWrapper dmlWrapper = new DmlWrapper();  
-        list<Account> listAccCreateAddr = new list<Account>();      
+        List<Account> ListAccCreateAddr = new List<Account>();      
+        List<Account> ListAccsAddrInfoCleared = new List<Account>();
        
-        integer i = -1;        
+        Integer i = -1;        
         for (SObject so : listNew) {
             Account acc = (Account)so;
             i++;
-
-            // BEFORE INSERT
-            if (triggerAction == TDTM_Runnable.Action.BeforeInsert) {
-            }
-            
-            // BEFORE UPDATE
-            if (triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
-            }
             
             // AFTER INSERT
             if (triggerAction == TDTM_Runnable.Action.AfterInsert) {
                 // for new accounts that are using address management, create the address object.
                 if (UTIL_CustomSettingsFacade.getSettings().Accounts_Addresses_Enabled__c != null && acc.RecordTypeId != null
                 && UTIL_CustomSettingsFacade.getSettings().Accounts_Addresses_Enabled__c.contains(acc.RecordTypeId)) {
-                    if (isAccountAddressSpecified(acc))
-                        listAccCreateAddr.add(acc);
+                    if (!ADDR_Addresses_UTIL.isAccAddressEmpty(acc))
+                        ListAccCreateAddr.add(acc);
                 }
             }
     
             // AFTER UPDATE
-            if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {
-                Account accOld = (Account)listOld[i];
-                // note that we detect whether the address change is coming from the address trigger,
-                // in which case we don't want to create a new address.
-                if (ADDR_Addresses_TDTM.hasRunAddrTrigger) {
-                    continue;
-                }
-                if (UTIL_CustomSettingsFacade.getSettings().Accounts_Addresses_Enabled__c != null && acc.RecordTypeId != null
-                && UTIL_CustomSettingsFacade.getSettings().Accounts_Addresses_Enabled__c.contains(acc.RecordTypeId)) {
-                    // if the address changed, remember the account we want to add a new address for    
-                    if (isAccountAddressChanged(acc, accOld)) 
-                        listAccCreateAddr.add(acc);
+            if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {  
+                // if address change is coming from the address trigger we don't want to create a new address   
+                if (!ADDR_Addresses_TDTM.hasRunAddrTrigger) {
+	                if (UTIL_CustomSettingsFacade.getSettings().Accounts_Addresses_Enabled__c != null && acc.RecordTypeId != null
+	                && UTIL_CustomSettingsFacade.getSettings().Accounts_Addresses_Enabled__c.contains(acc.RecordTypeId)) {
+	                    Account accOld = (Account)listOld[i];
+	                    // if the address info has been removed
```

Comment:
Nesting too many `IF` statements affects readability and makes maintenance more difficult. I suggest reviewing the design to simplify the flow of this code block.